src/main/java/com/ss/training/lms/controller/AdminBookController.java

```diff
@@ -0,0 +1,186 @@
+package com.ss.training.lms.controller;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import com.ss.training.lms.entity.Author;
+import com.ss.training.lms.entity.Book;
+import com.ss.training.lms.entity.BookAuthor;
+import com.ss.training.lms.entity.BookGenre;
+import com.ss.training.lms.entity.Genre;
+import com.ss.training.lms.service.admin.AdminAuthorService;
+import com.ss.training.lms.service.admin.AdminBookService;
+import com.ss.training.lms.service.admin.AdminGenreService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Trevor Huis in 't Veld
+ */
+@RestController
+@RequestMapping(path="lms/admin/book")
+public class AdminBookController {
+
+    @Autowired
+    AdminBookService bookService;
+
+    @Autowired
+    AdminGenreService genreService;
+
+    @Autowired
+    AdminAuthorService authorService;
+
+    /**
+	 * @param book
+	 * @return
+	 */
+	@PostMapping
+	public ResponseEntity<Book> createBook(@RequestBody Book book) {
+        HttpStatus status = HttpStatus.BAD_REQUEST;
+        if (book == null || book.getTitle() == null || book.getTitle().length() > 45 || book.getPublisherId() == null)
```

Comment:
The schema does allow a book to have a null publisher ID, but that doesn't mean our application necessarily has to allow it. For consistency, though, if a publisher ID is required here, it should also be required in the update method.